test/app_automate_driver_test.rb

```diff
@@ -0,0 +1,148 @@
+
+class AppAutomateDriverTest < Test::Unit::TestCase
+
+  USERNAME = "Username"
+  ACCESS_KEY = "Access_key"
+  LOCAL_ID = "Local_id"
+  APP_LOCATION = "App_location"
+  TEST_APP_URL = "Test_app_url"
+  TARGET_DEVICE = "ANDROID_9"
+  LOCAL_TUNNEL_COMMAND = "/BrowserStackLocal -d start --key #{ACCESS_KEY} --local-identifier #{LOCAL_ID} --force-local"
+
+  def test_defaults
+    AppAutomateDriver.any_instance.stubs(:upload_app).returns(TEST_APP_URL)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION)
+
+    assert_equal('errors', driver.caps[:'browserstack.console'])
+    assert_equal(LOCAL_ID, driver.caps[:'browserstack.localIdentifier'])
+    assert_equal('true', driver.caps[:'browserstack.local'])
+    assert_equal('true', driver.caps[:'browserstack.networkLogs'])
+    assert_equal(TEST_APP_URL, driver.caps[:app])
+
+    Devices::DEVICE_HASH[TARGET_DEVICE].each do |key, value|
+      assert_equal(value, driver.caps[key.to_sym])
+    end
+  end
+
+  def test_overridden_locator
+    AppAutomateDriver.any_instance.stubs(:upload_app).returns(TEST_APP_URL)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION, :accessibility_id)
+
+    assert_equal('errors', driver.caps[:'browserstack.console'])
+    assert_equal(LOCAL_ID, driver.caps[:'browserstack.localIdentifier'])
+    assert_equal('true', driver.caps[:'browserstack.local'])
+    assert_equal('true', driver.caps[:'browserstack.networkLogs'])
+    assert_equal(TEST_APP_URL, driver.caps[:app])
+
+    Devices::DEVICE_HASH[TARGET_DEVICE].each do |key, value|
+      assert_equal(value, driver.caps[key.to_sym])
+    end
+  end
+
+  def test_upload_app_success
+    json_response = JSON.dump({
+      :app_url => TEST_APP_URL
+    })
+    expected_command = %(curl -u "#{USERNAME}:#{ACCESS_KEY}" -X POST "https://api-cloud.browserstack.com/app-automate/upload" -F "file=@#{APP_LOCATION}")
+    AppAutomateDriver.any_instance.stubs(:`).with(expected_command).returns(json_response)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION)
+    assert_equal(TEST_APP_URL, driver.caps[:app])
+  end
+
+  def test_upload_app_error
+    json_response = JSON.dump({
+      :error => "Error"
+    })
+    expected_command = %(curl -u "#{USERNAME}:#{ACCESS_KEY}" -X POST "https://api-cloud.browserstack.com/app-automate/upload" -F "file=@#{APP_LOCATION}")
+    AppAutomateDriver.any_instance.stubs(:`).with(expected_command).returns(json_response)
+    assert_raise(RuntimeError, "BrowserStack upload failed due to error: Error") do
+      AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION)
+    end
+  end
+
+  def test_click_element_defaults
+    AppAutomateDriver.any_instance.stubs(:upload_app).returns(TEST_APP_URL)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION)
+
+    mocked_element = mock('object')
```

Comment:
It looks like you changed other lines to use `mock('element')` but forgot to change this line.